src/main/java/com/puppycrawl/tools/checkstyle/api/RootModule.java

```diff
@@ -39,10 +41,23 @@ public interface RootModule extends Configurable {
      * @param files the list of files to be audited.
      * @return the total number of audit events with error severity found
      * @throws CheckstyleException if error condition within Checkstyle occurs
+     * @deprecated use {@link #process(List)}
      * @see #destroy()
      */
     int process(List<File> files) throws CheckstyleException;
 
+    /**
+     * Processes a set of files.
+     * Once this is done, it is highly recommended to call for
+     * the destroy method to close and remove the listeners.
+     *
+     * @param files the list of files to be audited.
+     * @return the total number of audit events with error severity found
+     * @throws CheckstyleException if error condition within Checkstyle occurs
+     * @see #destroy()
+     */
+    int process(Collection<Path> files) throws CheckstyleException;
```

Comment:
```suggestion
    int process(List<Path> files) throws CheckstyleException;
```

Using the same type again is not allowed.

A higher-level interface is better, in my opinion. Why shouldn't we accept a `java.util.Set`, for example? Using `java.util.Collection` might be a cleaner and more flexible API anyway.

<img width="1068" alt="image" src="https://github.com/user-attachments/assets/666df3fc-9605-4b4e-ada2-7cb94f465555" />

Reply:
breaking API name, would have same type but I do not suggest this idea:

<img width="675" alt="image" src="https://github.com/user-attachments/assets/b309a3e4-d1a2-482e-9654-ef4383991b61" />

Reply:
how should we deal with this @romani @nrmancuso ?